asm/control-plane-revision/crd_control_plane_revision.yaml

```diff
@@ -0,0 +1,103 @@
+---
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: (unknown)
+  creationTimestamp: null
+  name: controlplanerevisions.mesh.cloud.google.com
+spec:
+  group: mesh.cloud.google.com
+  names:
+    kind: ControlPlaneRevision
+    listKind: ControlPlaneRevisionList
+    plural: controlplanerevisions
+    singular: controlplanerevision
+  scope: Namespaced
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      description: ControlPlaneRevision is the Schema for the ControlPlaneRevision
+        API
+      properties:
+        apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+          type: string
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          description: ControlPlaneRevisionSpec defines the desired state of ControlPlaneRevision
+          properties:
+            channel:
+              description: ReleaseChannel determines the aggressiveness of upgrades.
+              enum:
+                - regular
```

Comment:
Is this a direct copy? IIUC we have only allowed for regular right now and will be adding rapid/stable soon